### PR TITLE
Add server setting for resource timeouts

### PR DESCRIPTION
**Pull request by eemcmullan**
Thank you for contributing to Velero!

# Please add a summary of your change
Added a server config setting to use for any default timeouts.

# Does your change fix a particular issue?

Fixes #(issue)
#5604 

# Please indicate you've done the following:

- [ x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [x ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/restore/restore.go [hunk 1]
```diff
@@ -834,13 +839,13 @@ func (ctx *restoreContext) shouldRestore(name string, pvClient client.Dynamic) (
 
 // crdAvailable waits for a CRD to be available for use before letting the
 // restore continue.
-func (ctx *restoreContext) crdAvailable(name string, crdClient client.Dynamic) (bool, error) {
+func (ctx *restoreContext) crdAvailable(name string, crdClient client.Dynamic, resourceTimeout time.Duration) (bool, error) {
 	crdLogger := ctx.log.WithField("crdName", name)
 
 	var available bool
 	// Wait 1 minute rather than the standard resource timeout, since each CRD
 	// will transition fairly quickly.
```
Comment (blackpiglet) [inline pkg/restore/restore.go:847]: I think changing the timeout to the resourceTimeout should be safe, but also need to remove the comments here.

**Comment by blackpiglet** [comment]
Please add a changelog file too.

**Review by sseago — DISMISSED** [review]

File: pkg/cmd/server/server.go [hunk 2]
```diff
@@ -223,6 +226,7 @@ func NewCommand(f client.Factory) *cobra.Command {
 	command.Flags().DurationVar(&config.garbageCollectionFrequency, "garbage-collection-frequency", config.garbageCollectionFrequency, "How often garbage collection is run for expired backups.")
 	command.Flags().BoolVar(&config.defaultVolumesToFsBackup, "default-volumes-to-fs-backup", config.defaultVolumesToFsBackup, "Backup all volumes with pod volume file system backup by default.")
 	command.Flags().StringVar(&config.uploaderType, "uploader-type", config.uploaderType, "Type of uploader to handle the transfer of data of pod volumes")
+	command.Flags().DurationVar(&config.resourceTimeout, "resource-timeout", config.resourceTimeout, "How long to wait for certain resource processes before timeout.")
```
Comment (reasonerjt) [inline pkg/cmd/server/server.go:229]: Please clarify in the help text what the default value will be if this is not set, and also the relationship with other "timeout" param like `defaultCSISnapshotTimeout `, `resourceTerminatingTimeout`, `podVolumeOperationTimeout `

Comment (sseago) [reply pkg/cmd/server/server.go:229]: The default value should be listed here in the help text, I'd agree. I'm not 100% sure what we'd want to say about the relationship with other parameters, but maybe something like this:
"The default value is 5 minutes (or 10?). This parameter covers several cases involving resource creation or readiness that are not covered by the other more specific timeout parameters."

Comment (shubham-pampattiwar) [reply pkg/cmd/server/server.go:229]: +1 on stating the default value in `velero server help`

Comment (eemcmullan) [reply pkg/cmd/server/server.go]: updated

**Review by sseago — DISMISSED** [review]

File: pkg/restore/restore.go [hunk 3]
```diff
@@ -834,13 +839,12 @@ func (ctx *restoreContext) shouldRestore(name string, pvClient client.Dynamic) (
 
 // crdAvailable waits for a CRD to be available for use before letting the
 // restore continue.
-func (ctx *restoreContext) crdAvailable(name string, crdClient client.Dynamic) (bool, error) {
+func (ctx *restoreContext) crdAvailable(name string, crdClient client.Dynamic, resourceTimeout time.Duration) (bool, error) {
```
Comment (kaovilai) [inline pkg/restore/restore.go:842]:
```suggestion
func (ctx *restoreContext) crdAvailable(name string, crdClient client.Dynamic) (bool, error) {
```
restoreContext already have resourceTimeout within its structs. Why is there a need to pass this function parameter? restoreContext should've already been set with this value.

File: pkg/controller/backup_controller.go [hunk 4]
```diff
@@ -993,7 +996,7 @@ func (c *backupController) waitVolumeSnapshotReadyToUse(ctx context.Context,
 // change DeletionPolicy to Retain before deleting VS, then change DeletionPolicy back to Delete.
 func (c *backupController) deleteVolumeSnapshot(volumeSnapshots []snapshotv1api.VolumeSnapshot,
 	volumeSnapshotContents []snapshotv1api.VolumeSnapshotContent,
-	logger logrus.FieldLogger) {
+	logger logrus.FieldLogger, timeout time.Duration) {
```
Comment (kaovilai) [inline pkg/controller/backup_controller.go:999]:
```suggestion
	logger logrus.FieldLogger) {
```
backupController already have resourceTimeout within its structs. Why is there a need to pass this function parameter? backupController should've already been set with this value.

**Review by kaovilai — CHANGES_REQUESTED** [review]

**Review by kaovilai — DISMISSED** [review]

**Comment by blackpiglet** [comment]
@eemcmullan 
Hi, this PR looks good, but there are some conflicts due to the recent PRs merge. Could you please resolve the conflicts?

**Review by sseago — APPROVED** [review]

**Review by blackpiglet — DISMISSED** [review]

**Comment by blackpiglet** [comment]
@eemcmullan 
Some PRs have been merged recently. I tried to resolve the new conflict on the web editor, but the code's indent is messed up.
Could you take a look?

**Review by sseago — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**